app/controllers/your-psp/your-psp-tasks.service.test.js

```diff
@@ -204,5 +204,174 @@ describe('Stripe task list', () => {
       const taskList = getStripeTaskList(activeCredential, account)
       expect(stripeTaskListIsComplete(taskList)).to.equal(false)
     })
+
+    it('should correctly calculate when zero tasks are complete ', () => {
+      const account = gatewayAccountFixtures.validGatewayAccount({
+        gateway_account_credentials: [
+          { state: 'ACTIVE', payment_provider: 'stripe', id: 100 }
+        ]
+      })
+      account.connectorGatewayAccountStripeProgress = {
+        bankAccount: false,
+        director: false,
+        vatNumber: false,
+        companyNumber: false,
+        responsiblePerson: false,
+        organisationDetails: false,
+        governmentEntityDocument: false
+      }
+
+      const activeCredential = getCurrentCredential(account)
+      const taskList = getStripeTaskList(activeCredential, account)
+      console.log(1111111, (stripeTaskListNumberOftasksComplete(taskList)))
+      expect(stripeTaskListNumberOftasksComplete(taskList).no_of_tasks_completed).to.equal('0')
+      expect(stripeTaskListNumberOftasksComplete(taskList).total_number_of_tasks).to.equal('7')
+    })
+
+    it('should correctly calculate when 1 task is complete ', () => {
+      const account = gatewayAccountFixtures.validGatewayAccount({
+        gateway_account_credentials: [
+          { state: 'ACTIVE', payment_provider: 'stripe', id: 100 }
+        ]
+      })
+      account.connectorGatewayAccountStripeProgress = {
+        bankAccount: true,
+        director: false,
+        vatNumber: false,
+        companyNumber: false,
+        responsiblePerson: false,
+        organisationDetails: false,
+        governmentEntityDocument: false
+      }
+      const activeCredential = getCurrentCredential(account)
+      const taskList = getStripeTaskList(activeCredential, account)
+      expect(stripeTaskListNumberOftasksComplete(taskList).no_of_tasks_completed).to.equal('1')
+      expect(stripeTaskListNumberOftasksComplete(taskList).total_number_of_tasks).to.equal('7')
+    })
+
+    it('should correctly calculate when 2 tasks are complete ', () => {
+      const account = gatewayAccountFixtures.validGatewayAccount({
+        gateway_account_credentials: [
+          { state: 'ACTIVE', payment_provider: 'stripe', id: 100 }
+        ]
+      })
+      account.connectorGatewayAccountStripeProgress = {
+        bankAccount: true,
+        director: true,
+        vatNumber: false,
+        companyNumber: false,
+        responsiblePerson: false,
+        organisationDetails: false,
+        governmentEntityDocument: false
+      }
+      const activeCredential = getCurrentCredential(account)
+      const taskList = getStripeTaskList(activeCredential, account)
+      expect(stripeTaskListNumberOftasksComplete(taskList).no_of_tasks_completed).to.equal('2')
+      expect(stripeTaskListNumberOftasksComplete(taskList).total_number_of_tasks).to.equal('7')
+    })
+
+    it('should correctly calculate when 3 tasks are complete ', () => {
```

Comment:
Don't think we need individual tests for all tasks. We can have 3 test cases covering

- all tasks completed
- none of the tasks completed
- few tasks completed